### DISCUSSION TITLE
Getting Started Tutorial

### DISCUSSION THREAD

**Discussion by ubaidsk** [WebAssembly/binaryen#4593] [General]
Do we have any getting started tutorial for the C++ API? like a blog/readme/article/website/youtube-video etc.?

**Comment by tlively** [comment]
No, although that would be nice to have. At least a section in the README about how to link against libbinaryen.so would be a good place to start, and from there it wouldn't be hard to add a simple walkthrough of a "hello world"-type application that maybe just loads and prints a module.

We do have the test/example directory which might be useful to look at, but in practice those files are actually tests, so they're not really designed to teach the API.

**Reply by ubaidsk** [reply]
@tlively Thank you for the response! I will look at the examples in the test/example directory.

By the way, just like we have a single header C API, do we also have a single header C++ API?

**Reply by tlively** [reply]
No, the C++ API includes all of the headers you can find in the source tree. One of these days I would like to at least move them all to a separate `include/` directory.

**Reply by ubaidsk** [reply]
Ok, Thank you for the response!

**Reply by kripken** [reply]
Btw, while `test/example/` are tests, they should also be useful as examples. At least I've been thinking of them that way. Hence the name :smile:

**Comment by ubaidsk** [comment]
Is there a way to name the variables being used in the WebAssembly Text format?

**Reply by tlively** [reply]
Yes, check out `Function::setLocalName` declared in src/wasm.h.

**Reply by ubaidsk** [reply]
Thank you @tlively !
